### PR TITLE
Add NumFOCUS sponsored project section

### PR DESCRIPTION
**Pull request by jrbourbeau**
This PR adds a "Sponsored By" section for NumFOCUS. Definitely open to suggestions if others would like to go about including this in a different way. The only restrictions are we need to include the "Dask is a fiscally sponsored project..." text as well as the fiscally sponsored project image.

ref: https://github.com/dask/dask/issues/5082

### CODE REVIEW

**Comment by jrbourbeau** [comment]
Here's a screenshot with the changes in the initial commit:

Screen Shot 2019-07-29 at 2 39 18 PM

File: index.html [hunk 1]
```diff
@@ -430,6 +430,22 @@ <h4>tsfresh</h4>
 
       </div>
 
+      <section id="sponsors">
+        <div class="container sponsors">
+          <h2>Sponsored By</h2>
+            <p>Dask is a fiscally sponsored project of NumFOCUS, a nonprofit dedicated
+            to supporting the open source scientific computing community. If you like
+            Dask and want to support our mission, please consider making a
+            <a href="#">donation</a> to support our efforts.</p>
```
Comment (jrbourbeau) [inline index.html]: Note that I'll update this with a dask-specific donation link

**Comment by mrocklin** [comment]
I'm somewhat in favor of a less prominent placement in a few ways.

Currently NumFOCUS doesn't play a prominent role in this project, certainly less than the insittutions all sharing a line just below this placement.

Instead, I'll recommend a slimmer line after the "Supported By" section that includes a description and a scaled down logo on a single line.  I also don't think that we should use the term "Sponsored By", as this implies to most people that NumFOCUS pays for Dask development.  I'd rather focus more on the legal relationship that NumFOCUS brings rather than any financial implication.

Similarly, I'm not super excited about the donation link.  Putting on my sales/partnership hat for a moment, asking for a donation in this way sort of changes the conversation to "give us $100" and away from "You could probably accomplish a lot internally by investing $200k annually in this project".  Of course, collecting small-dollar donations is always welcome, we might eventually be able to pay for someone's flight or conference attendance, but I'm in favor of not displaying this as prominently on the main page.

**Comment by mrocklin** [comment]
(Also, goes without saying, but thanks for doing this work)

**Comment by mrocklin** [comment]
Bokeh places this more as a footer, which seems sensible to me: https://bokeh.org/

**Comment by jrbourbeau** [comment]
Thanks for the thoughtful feedback @mrocklin. I pushed a commit that moves the NumFOCUS information to the footer a la bokeh. Currently, it looks like:

Screen Shot 2019-07-29 at 9 12 45 PM

**Comment by mrocklin** [comment]
Personally I think that that looks great.

**Comment by TomAugspurger** [comment]
LGTM too. Thanks James.

**Comment by jrbourbeau** [comment]
Great! Thanks for taking a look. I'll follow up with NumFOCUS about getting a donation link

**Comment by jrbourbeau** [comment]
The donation link has been added. Planning to merge soon if no other comments

**Pull request merged**